### PR TITLE
Restore EEPROM support

### PR DESCRIPTION
**Pull request by nseidle**
In [this commit](https://github.com/sparkfun/Arduino_Apollo3/commit/0f54d0f416eb60de86218833e87038456383ac13#diff-5c378d0844f0422d4d813eabe83f61ffce82014d8b8aa3e93ee35ce2ca14ca7bR27), EEPROM compilation was fixed. But later in [this commit](https://github.com/sparkfun/Arduino_Apollo3/commit/2de75c81921d2141a11c5c4e6bafe3baa2acc0d1#diff-5c378d0844f0422d4d813eabe83f61ffce82014d8b8aa3e93ee35ce2ca14ca7bR27), it was broken again.

This PR:

* Fixes the missing drivers for FlashIAPBlockDevice.h error
* Changes how write() works. Now uses an uint8_t based scratch to avoid writing corruption across 32-bit blocks.
* Adds back example 3 to demo all functions
* Correct typo in example 1 so that it appears in dropdown menu

Nothing about this PR has me worried. The only question I have is why the `includes.extra="-I{cores.path}/mbed-os/drivers/"` was removed. If this was an error, then I think this PR is good to go. If the removal was intentional because of some other issue then we need further review.

### CODE REVIEW

**Comment by Wenn0101** [comment]
Hey @nseidle 
The `includes.extra="-I{cores.path}/mbed-os/drivers/"` was removed because this folder also contains a SPI.h. This conflicted with the arduino core SPI.h and is what cause SPI compilation error in v2.0.4. To address this, I removed  `includes.extra="-I{cores.path}/mbed-os/drivers/"` again, but I changed our version of FlashIAPBlockDevice.h in mbed core to have a more explicit include path `#include "FlashIAP.h"` became `#include "drivers/FlashIAP.h"`. See [this commi](https://github.com/sparkfun/mbed-os-ambiq-apollo3/commit/6b6244e40c3eab581630b050f8b80f70bf08bbfc)t.
I thought I updated the submodules correctly so that this would work in the latest build.
Are you seeing the problem when you download from the arduino boards manager, or did you get it from git? If the latter, can you try running a `git submodule update --init --recursive` and see if this clears up the problem. I thought I double checked this before release, so I would be somewhat surprised if it was not working.

**Comment by nseidle** [comment]
Ah! That makes sense.

Yep, your update recommendation fixed the issue. I'm able to compile without the includes.extra. I've updated the PR. Let's let a few other folks kick the tires on release candidate but we should be good.

File: libraries/EEPROM/src/EEPROM.h [hunk 1]
```diff
@@ -40,16 +46,17 @@ class EEPROMClass : public FlashIAPBlockDevice {
...
     void write(int idx, uint8_t* data, uint32_t size){
-        mbed::bd_size_t scratch_size = (_cfg.sram_bytes+3)/4;
-        uint32_t scratch[scratch_size];
+        mbed::bd_size_t scratch_size = _cfg.sram_bytes;
+        uint8_t scratch[scratch_size];
+
         FlashIAPBlockDevice::read((uint8_t*)scratch, 0, _cfg.sram_bytes);   // keep all of flash in sram in case we need to erase
-		if(memcmp((void*)(((uint8_t*)scratch) + idx), data, size)){         // compare desired data (data) to existing information in flash (scratch)
+
+        if(memcmp((void*)(((uint8_t*)scratch) + idx), data, size)){         // compare desired data (data) to existing information in flash (scratch)
```
Comment (Wenn0101) [inline libraries/EEPROM/src/EEPROM.h:54]: This casting seems unnecessary now that scratch is a uint8_t.

**Review by Wenn0101 — APPROVED** [review]

**Comment by Wenn0101** [comment]
I just noticed, after approving. Can we change this to merge into dev instead of master?

**Comment by nseidle** [comment]
I based off of rc but I suspect Windows GUI defaults to opening PR on main. Should be rc now.

**Comment by Wenn0101** [comment]
rc is actually an automatically generated branch that is based off dev. It takes the info in dev, and uses it to to build the precompiled libraries of the mbed submodule. Changed the base to dev, since I noticed I have that power too :D.

I'll take one more look at this tomorrow before merging it in.

**Review by Wenn0101 — APPROVED** [review]

**Pull request merged**